src/lib/mutate-request.ts

```diff
@@ -11,19 +11,19 @@ export class PeprMutateRequest<T extends KubernetesObject> {
   Raw: T;
   #input: AdmissionRequest<T>;
 
-  get PermitSideEffects() {
+  get PermitSideEffects(): boolean {
     return !this.#input.dryRun;
   }
 
-  get IsDryRun() {
+  get IsDryRun(): boolean | undefined {
     return this.#input.dryRun;
   }
 
-  get OldResource() {
+  get OldResource(): T | undefined {
```

Comment:
```suggestion
  get OldResource(): KubernetesObject | undefined {
```

I think we can declare the type at compile-time instead of run-time. The constructor for this class uses `AdmissionRequest<T extends KubernetesObject>`. Currently, we're using a [generic](https://www.typescriptlang.org/docs/handbook/2/generics.html) which we might not actually need.

Reply:
good call out!